### PR TITLE
ripgrep: 13.0.0 -> 14.0.1

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/text/ripgrep/default.nix [hunk 1]
```diff
@@ -11,29 +10,32 @@
...
     repo = pname;
     rev = version;
-    sha256 = "0pdcjzfi0fclbzmmf701fdizb95iw427vy3m1svy6gdn2zwj3ldr";
+    sha256 = "sha256-yVRjOwifxjxuvuwF2d7QCNb7PaT3ELoGP34T2RE1ZVY=";
   };
 
-  cargoSha256 = "1kfdgh8dra4jxgcdb0lln5wwrimz0dpp33bq3h7jgs8ngaq2a9wp";
+  cargoSha256 = "sha256-c4rJYZkAa8vqw3/ccOjGMoyzqq7CVDAMOme9/ORmx9M=";
 
-  nativeBuildInputs = [ asciidoctor installShellFiles ]
```
Comment (philiptaron) [inline pkgs/tools/text/ripgrep/default.nix:25]: Love it when a build dependency vanishes.

File: pkgs/tools/text/ripgrep/default.nix [hunk 2]
```diff
@@ -11,29 +10,32 @@
...
-    installManPage $releaseDir/build/ripgrep-*/out/rg.1
+    $out/bin/rg --generate man > rg.1
+    installManPage rg.1
 
-    installShellCompletion $releaseDir/build/ripgrep-*/out/rg.{bash,fish}
-    installShellCompletion --zsh complete/_rg
+    installShellCompletion --cmd rg \
+      --bash <($out/bin/rg --generate complete-bash) \
+      --fish <($out/bin/rg --generate complete-fish) \
+      --zsh <($out/bin/rg --generate complete-zsh)
```
Comment (philiptaron) [inline pkgs/tools/text/ripgrep/default.nix:38]: Sweet, saw this in the release notes.

**Review by philiptaron — APPROVED** [review]
Looks great!

**Comment by techknowlogick** [comment]
Result of `nixpkgs-review pr 270281` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>bat-extras.batgrep</li>
    <li>lunarvim</li>
    <li>openvscode-server</li>
    <li>repgrep</li>
    <li>ripgrep</li>
    <li>ripgrep-all</li>
    <li>spacevim</li>
    <li>yazi</li>
  </ul>
</details>

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**